core/src/main/java/io/questdb/cairo/wal/ApplyWal2TableJob.java

```diff
@@ -454,14 +472,17 @@ private long processWalCommit(
                                 seqTxn,
                                 commitToTimestamp
                         );
+                        telemetryFacade.store(APPLIED_TXN, writer.getTableToken(), walId, seqTxn, rowsAdded, referenceTimestamp);
```

Comment:
This is not the time it takes to apply transaction, it is the time the transaction was waiting before the application started. The waiting time is not relevant to the transaction type whether it is data or sql transaction.

I suggest to log the first even as something like  WAL apply WAIT or QUEUE. It is same for transaction types.
Second even when application is done should be transaction specific, like `WAL_APPLY_TXN_DATA`. 

Something like 

```java
                        referenceTimestamp = telemetryFacade.store(WAL_APPLY_TXN_DEQUEUE, writer.getTableToken(), walId, seqTxn, -1L, commitTimestamp);
                        final long rowsAdded = writer.processWalData(
                                walPath,
                                !dataInfo.isOutOfOrder(),
                                dataInfo.getStartRowID(),
                                dataInfo.getEndRowID(),
                                dataInfo.getMinTimestamp(),
                                dataInfo.getMaxTimestamp(),
                                dataInfo,
                                seqTxn,
                                commitToTimestamp
                        );
                        telemetryFacade.store(WAL_APPLY_TXN_DATA, writer.getTableToken(), walId, seqTxn, rowsAdded, referenceTimestamp);
```

The event `WAL_APPLY_TXN_DEQUEUE` is always the first one to log for every transaction type.

Reply:
Fixed in https://github.com/questdb/questdb/pull/2930/commits/6ecb9fddca771bfe59923c88c596b964b9e01879